### PR TITLE
cortex: Define Azure object storage ruler args

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does**:
Define ruler object storage arguments for Azure.

**Which issue(s) this PR fixes**:

**Checklist**
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: cortex/config.libsonnet [hunk 1]
```diff
@@ -258,7 +258,11 @@
           'ruler-storage.s3.region': $._config.aws_region,
           'ruler-storage.s3.bucket-name': $._config.ruler_storage_bucket_name,
         },
-        azure: {},
+        azure: {
+          'ruler-storage.azure.container-name': '%(cluster)s-%(namespace)s-ruler' % $._config,
+          'ruler-storage.azure.account-name': '$(BLOCKS_STORAGE_AZURE_ACCOUNT_NAME)',
+          'ruler-storage.azure.account-key': '$(BLOCKS_STORAGE_AZURE_ACCOUNT_KEY)',
```
Comment (pracucci) [inline cortex/config.libsonnet:264]: These should all be configurable. Not autogenerated (and what's `BLOCKS_STORAGE_AZURE_ACCOUNT_NAME`?):

Comment (aknuds1) [reply cortex/config.libsonnet:264]: @pracucci OK, looking into it.

Comment (jdbaldry) [reply cortex/config.libsonnet:264]: For my understanding, when you say "should be configurable", do you mean that they should exist as fields in the `_config` object?

The values `BLOCKS_STORAGE_AZURE.*` look like Kubernetes environment variables but I'm not sure how they are getting to the Pod spec. Is there something special in Azure or is this tbd?

Comment (aknuds1) [reply cortex/config.libsonnet]: I've pushed a revision parameterizing the fields instead.

File: cortex/config.libsonnet [hunk 2]
```diff
@@ -258,7 +260,11 @@
           'ruler-storage.s3.region': $._config.aws_region,
           'ruler-storage.s3.bucket-name': $._config.ruler_storage_bucket_name,
         },
-        azure: {},
+        azure: {
+          'ruler-storage.azure.container-name': $._config.ruler_storage_bucket_name,
+          'ruler-storage.azure.account-name': $._config.ruler_storage_azure_account_name,
+          'ruler-storage.azure.account-key': $._config.ruler_storage_azure_account_key,
```
Comment (jdbaldry) [inline cortex/config.libsonnet:266]: You were too fast for me! Thanks for fixing the typo :)

**Review by jdbaldry — APPROVED** [review]
Looks like all the flags we need. I've double checked for any typos and it looks good to me

File: cortex/config.libsonnet (hunk 2 above)
Comment (aknuds1) [reply cortex/config.libsonnet:266]: @jdbaldry yeah I made a bit of a mess at first, sorry :)

Comment (jdbaldry) [reply cortex/config.libsonnet:266]: Not problem at all!

**Comment by aknuds1** [comment]
I've tested the latest revision FWIW, and it looks good.

**Review by pracucci — APPROVED** [review]

**Pull request merged**